services/s3/src/main/resources/codegen-resources/customization.config

```diff
@@ -6,6 +6,17 @@
     "Error": "S3Error",
     "Object": "S3Object"
   },
+  "shapeAdditions": {
+    "shapes":{
+      "PartType":{
+        "type":"string",
```

Comment:
Test cases if the  ` "type":"string"` for below cases
1.         "type":null
2.          "type": CustomShape
3.         "type":Complex shapes like List Maps etc .

Reply:
+1 let's add some codegen tests to show how the generated code looks like, at least for the same type.

Reply:
added unit tests and codegen tests